python/enthic/compute_stats.py

```diff
@@ -0,0 +1,211 @@
+        elif child_missing_count == 1:
+            for child_name in item['children']:
+                child_data = item['children'][child_name]['data']
+                if math.isnan(child_data["value"]):
+                    child_data['computedValue'] = (value - computed_sum) / child['sign']
+                    child_data['value'] = child_data['computedValue']
+                    child_data['status'] = "computed"
+                    item['data']['status'] = "checked"
+                    break
+
+        # If official value match computed value by adding all children's value (no substraction) with less than 0.5% error
+        elif abs((computed_sum_without_sign - value) / value) < relative_error or abs(computed_sum_without_sign - value) < absolute_error:
+            item['data']['status'] = "checked"
+            # Fix children sign and/or set to zero missing values if any
+            for child_name in item['children']:
+                flip_sign(item['children'][child_name])
+                set_to_zero_computed(item['children'][child_name])
+        elif abs((computed_sum - value) / value) > 100:
+            item['data']['status'] = "error"
+        else:
+            item['data']['status'] = "error"
+
+    if computed_sum != item['data']['value']:
+        item['data']['computedValue'] = computed_sum_from_computed
+
+def set_to_zero_computed(item):
+    if math.isnan(item['data']['value']) and ('computedValue' not in item['data'] or item['data']['computedValue'] == 0):
+        item['data']['value'] = 0
+        item['data']['status'] = "computed"
+        if 'children' in item:
+            for child_name in item['children']:
+                set_to_zero_computed(item['children'][child_name])
+
+def flip_sign(item):
+    if item.sign == -1:
+        item['data']['value'] = -item['data']['value']
+        item['data']['status'] = "signFlipped"
+
+def compute_annual_share_score(tree):
+    # Retrieve needed values
+    participation = tree['children']['ParticipationSalariesAuxResultats']['data']['value']
+    impot = tree['children']['ImpotsSurLesBenefices']['data']['value']
+    resultat_financier = tree['children']['ResultatExceptionnel']['data']['value']
+    resultat_exceptionnel = tree['children']['ResultatAvantImpot']['children']['ResultatFinancier']['data']['value']
+    resultat_exploitation = tree['children']['ResultatAvantImpot']['children']['ResultatExploitation']['data']['value']
+
+    # If any values unknown, cannot compute score
+    if any(math.isnan(value) for value in [participation, impot, resultat_financier, resultat_exceptionnel, resultat_exploitation]):
+        return float('nan')
+
+    # Compute score if possible
+    shared_part = participation + impot
+    resultat_capitaux = resultat_financier + resultat_exceptionnel
+    if shared_part >= 0:
+        if resultat_capitaux <= 0:
+            return shared_part / resultat_exploitation
+
+    return float('nan')
+
+def compute_salary_scores(tree):
```

Comment:
Please add a properly restructured text docsrting documenting the code. You can use https://livesphinx.herokuapp.com/ to help formatting.